pkg/asset/manifests/azure/cluster.go

```diff
@@ -267,7 +267,11 @@ func getNextAvailableIP(ctx context.Context, installConfig *installconfig.Instal
 	if err != nil {
 		return "", fmt.Errorf("failed to get azure ip availability: %w", err)
 	}
-	if *availableIP.Available {
+	if availableIP == nil {
+		return "", fmt.Errorf("failed to get available IP in given machine network: %w", err)
```

Comment:
@jinyunma traced the issue back to insufficient permissions
It is strange that the client.CheckIPAddressAvailability does not return an error with inadequate permissions but looking at the code 

https://github.com/openshift/installer/blob/d5a8a36ab553056b5567ecba8a1d78686c53e505/pkg/asset/installconfig/azure/client.go#L75

All of the errors there and here appear to be appropriately checked.

Can you update the error message to indicate this might be an issue? Also the error in this case will be nil, so we should not return it.

Something like
```suggestion
		return "", errors.New("failed to get available IP in given machine network: this error may be caused by lack of necessary permissions")
```

Reply:
Everything API related is weird at this point. Fixed